Decision records/01 - Use our own testing library for testing React codebases.md

```diff
@@ -0,0 +1,56 @@
+# We use @shopify/react-testing for testing React codebases
+
+## Date
+
+(retroactive) April 1, 2020
+
+## Contributors
+
+- @themallen (Mathew Allen)
+
+## Summary
+
+We use and maintain [`@shopify/react-testing`](https://github.com/Shopify/quilt/blob/master/packages/react-testing/README.md) for testing React applications at Shopify. Usage of supplemental tools for visual regression testing and to aid in manual functional testing is also encouraged where needed.
+
+## Problem space
+
+There are a number of testing libraries available for React applications. The most popular of these focus on testing the behaviour of components in response to initial data and subsequent user interaction and the composition of components together. Testing of the actual visual output of a component is usually not included in scope of these libraries, or is encouraged to be done via snapshots of the DOM (which is highly [problematic](https://github.com/Shopify/web-foundation/blob/master/handbook/Decision%20records/03%20-%20We%20do%20not%20use%20Jest%20snapshot%20tests.md)).
+
+### Prior Art #1 - Enzyme
+
+[Enzyme](https://enzymejs.github.io/enzyme/) is a very popular testing library which offers many features such as JQuery style traversal of the React component tree, shallow rendering, and a variety of tools for simulating interactions with an application.
+
+Unfortunately, Enzyme has a number of downsides for our usecase.
+
+- It has frequently taken a long time to support new React features
```

Comment:
IIRC, Enzyme not supporting hooks was a big deal-breaker. I think it's worth highlighting that here.

Reply:
Yah, I think it would be good to provide a specific example for the record.